docsrc/workflow.rst

```diff
@@ -0,0 +1,193 @@
+
+    import os
+    from cmdstanpy import CmdStanModel
+
+    my_stanfile = os.path.join('.', 'my_model.stan')
+    my_model = CmdStanModel(stan_file=my_stanfile)
+    my_model.name
+    my_model.stan_file
+    my_model.exe_file
+    my_model.code()
+
+The method :meth:`cmdstanpy.CmdStanModel.compile` is used to compile the model as needed.
+When the argument ``force=True`` is present, CmdStanPy will always compile the model,
+even if the existing executable file is newer than the Stan program file.
+ 
+Model compilation is carried out via the GNU Make build tool.
+The CmdStan ``makefile`` contains a set of general rules which
+specify the dependencies between the Stan program and the
+Stan platform components and low-level libraries.
+Optional behaviors can be specified by use of variables
+which are passed in to the ``make`` command as name, value pairs.
+
+Model compilation is done in two steps:
+
+* The ``stanc`` compiler translates the Stan program to C++.
+* The C++ compiler compiles the generated code and links in
+  the necessary supporting libraries.
+
+Therefore, both the constructor and the ``compile`` method
+allow optional arguments ``stanc_options`` and ``cpp_options`` which
+specify options for each compilation step.
+Options are specified as a Python dictionary mapping
+compiler option names to appropriate values.
+
+To use Stan's 
+`parallelization <https://mc-stan.org/docs/cmdstan-guide/parallelization.html>`__
+features, Stan programs must be compiled with the appropriate C++ compiler flags.
+If you are running GPU hardware and wish to use the OpenCL framework to speed up matrix operations,
+you must set the C++ compiler flag **STAN_OPENCL**.
+For high-level within-chain parallelization using the Stan language `reduce_sum` function,
+it's necessary to set the C++ compiler flag **STAN_THREADS**.  While any value can be used,
+we recommend the value ``True``.
+
+For example, given Stan program named 'proc_parallel.stan', you can take
+advantage of both kinds of parallelization by specifying the compiler options when instantiating
+the model:
+
+.. code-block:: python
+
+    proc_parallel_model = CmdStanModel(
+        stan_file='proc_parallel.stan',
+        cpp_options={"STAN_THREADS": True, "STAN_OPENCL": True},
+    )
+
+
+Assemble input and initialization data
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+CmdStan is file-based interface, therefore all model input and
+initialization data must be supplied as JSON files, as described in the
+CmdStan User's Guide:
+<https://mc-stan.org/docs/cmdstan-guide/json.html>.
+
+CmdStanPy inference methods allow inputs and initializations
+to be specified as in-memory Python dictionary objects
+which are then converted to JSON via the utility method :func:`cmdstanpy.write_stan_json`.
+This method should be used to create JSON input files whenever
+these inputs contain either a collection compatible with
+numpy arrays or pandas.Series.
+
+
+Run the CmdStan inference engine
```

Comment:
We should consider linking to the example notebooks in these bullet points (see #431)